compiler/src/iree/compiler/Codegen/Common/VectorLayoutAnalysis.cpp

```diff
@@ -772,6 +793,11 @@ void enforcementTransferFunction(
     return;
   }
 
+  if (auto gather = dyn_cast<vector::GatherOp>(op)) {
+    enforceLayoutToGatherOp(gather, operandLattices, resultLattices, update);
+    return;
+  }
```

Comment:
We also need a propagation transfer function for vector.gather i think. You can look at other examples.